packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -0,0 +1,231 @@
+		>
+			<div> { __( 'Insert from URL' ) } </div>
+		</MenuItem>
+	);
+
+	let urlInputUIContent;
+	if ( showEditURLInput ) {
+		urlInputUIContent = ( <LinkEditor
+			onKeyDown={ stopPropagationRelevantKeys }
+			onKeyPress={ stopPropagation }
+			value={ mediaURLValue }
+			isFullWidthInput={ true }
+			hasInputBorder={ true }
+			onChangeInputValue={ ( url ) => ( setMediaURLValue( url ) ) }
+			onSubmit={ ( event ) => {
+				event.preventDefault();
+				onSelectURL( mediaURLValue );
+				setShowEditURLInput( ! showEditURLInput );
+				editMediaButtonRef.current.focus();
+			} }
+		/> );
+	} else {
+		urlInputUIContent = ( <LinkViewer
+			isFullWidth={ true }
+			className="editor-format-toolbar__link-container-content block-editor-format-toolbar__link-container-content"
+			url={ mediaURLValue }
+			onEditLinkClick={ () => ( setShowEditURLInput( ! showEditURLInput ) ) }
+		/> );
+	}
+
+	const urlInputUI = ( showURLInput && <div className="block-editor-media-flow__url-input">
+		{ urlInputUIContent }
+	</div> );
+
+	const onClose = () => {
+		editMediaButtonRef.current.focus();
+	};
+
+	const onClickOutside = () => {
+		return ( ) => {
+			setShowMediaReplaceOptions( false );
+		};
+	};
+
+	const editMediaButton = (
+		<MediaUploadCheck>
+			<MediaUpload
+				onSelect={ ( media ) => selectMedia( media ) }
+				onClose={ () => setShowMediaReplaceOptions( true ) }
+				allowedTypes={ allowedTypes }
+				render={ ( { open } ) => (
+					<>
+						<Toolbar className={ 'mediaflow-replace components-dropdown-menu components-toolbar' }>
+							<Button
+								ref={ editMediaButtonRef }
+								className={ 'components-button components-icon-button components-dropdown-menu__toggle' }
+								onClick={ () => {
+									setShowMediaReplaceOptions( ! showMediaReplaceOptions );
+								} }
+							>
+								<span className="components-dropdown-menu__label" > { name } </span>
+								<span className="components-dropdown-menu__indicator" />
+							</Button>
+							{ showMediaReplaceOptions &&
+								<Popover
+									onClickOutside={ onClickOutside() }
+									onClose={ onClose }
+								>
+									<>
+										<NavigableMenu>
+											<MenuItem
+												icon="admin-media"
+												onClick={ open }
+											>
+												{ __( 'Open Media Library' ) }
+											</MenuItem>
+											{ fileUploadButton }
+											{ URLButton }
+										</NavigableMenu>
+										{ urlInputUI }
+									</>
+								</Popover>
+							}
+						</Toolbar>
+					</>
+				) }
+			/>
+		</MediaUploadCheck>
+	);
+
+	return (
+		<>
+			{ editMediaButton }
+		</>
+	);
+};
+
+const applyWithSelect = withSelect( ( select ) => {
+	const { getSettings } = select( 'core/block-editor' );
+
+	return {
+		mediaUpload: getSettings().mediaUpload,
+	};
+} );
+
+/**
+ * @see https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/media-placeholder/README.md
+ */
+export default compose(
+	applyWithSelect,
+	withNotices,
```

Comment:
We are using withNotices, and we create notices when an upload error happens but we never show these errors, so when an upload fails the user does not understand why. I think the component should receive an onError prop passed by the blocks as MediaPlaceholder does and in that prop, the block creates the notice.

Reply:
I thought that is what 

```
const onUploadError = ( message ) => {
	noticeOperations.removeAllNotices();
	noticeOperations.createErrorNotice( message );
};
```

handles. I didn't sim. any upload failures tho :)

Mediaupload is wired to the handler above:

```
mediaUpload( {
	allowedTypes,
	filesList: files,
	onFileChange: setMedia,
	onUploadError,
} );
```

Reply:
ok @jorgefilipecosta I have followed your advice and now in case of error we end up on the media placeholder with the error displayed, which is the default way the image block shows upload errors. My mistake was that I thought without researching that notices are those HUD style notifications ... :)